test/spacebindingrequest/spacebindingrequest.go

```diff
@@ -34,3 +36,16 @@ func WithSpaceRole(spaceRole string) Option {
 		spaceBindingRequest.Spec.SpaceRole = spaceRole
 	}
 }
+
+func WithDeletionTimestamp() Option {
+	return func(spaceBindingRequest *toolchainv1alpha1.SpaceBindingRequest) {
+		now := metav1.NewTime(time.Now())
+		spaceBindingRequest.DeletionTimestamp = &now
+	}
+}
+
+func WithFinalizer() Option {
+	return func(spaceBindingRequest *toolchainv1alpha1.SpaceBindingRequest) {
+		spaceBindingRequest.Finalizers = append(spaceBindingRequest.Finalizers, toolchainv1alpha1.FinalizerName)
```

Comment:
what if the `spaceBindingRequest` object is not `nil` before applying the finalizer? Would that be a problem? If so:

```go
if spaceBindingRequest != nil {
	spaceBindingRequest.Finalizers = append(spaceBindingRequest.Finalizers, toolchainv1alpha1.FinalizerName)
} else {
	// Handle the case where the input spaceBindingRequest is nil, possibly with logging or an error return.
		}
```

Reply:
Unless I'm missing something, it cannot be nil, since those `Option` functions are being passed to the `NewSpaceBindingRequest` function, which takes care of creating the SpaceBindingRequest object: https://github.com/codeready-toolchain/host-operator/blob/master/test/spacebindingrequest/spacebindingrequest.go#L13-L19

Of course if somehow misused you could break them, but those are "only" used for unit testing, and we might want to modify all our `With...` helper in this case (also the ones in toolchain-common), since they all follow the same style, eg: https://github.com/codeready-toolchain/toolchain-common/blob/master/pkg/test/usersignup/usersignup.go